### PR TITLE
Remove onDelete from BuildDefinition

### PR DESCRIPTION
**Pull request by grouma**
First step towards https://github.com/dart-lang/build/issues/1427

We intend to hide the details of BuildDefinition from `watch_impl.dart`. This is a required first step.

### CODE REVIEW

File: build_runner/lib/src/generate/build_definition.dart [hunk 1]
```diff
@@ -442,8 +435,5 @@ class _Loader {
     }
   }
 
-  Future _delete(AssetId id, RunnerAssetWriter writer) {
-    _onDelete?.call(id);
-    return writer.delete(id);
-  }
+  Future _delete(AssetId id, RunnerAssetWriter writer) => writer.delete(id);
```
Comment (natebosch) [inline build_runner/lib/src/generate/build_definition.dart]: can we hand inline this?

File: build_runner/lib/src/generate/watch_impl.dart [hunk 2]
```diff
@@ -180,8 +203,10 @@ class WatchImpl implements BuildState {
   /// Only one build will run at a time, and changes are batched.
   ///
   /// File watchers are scheduled synchronously.
-  Stream<BuildResult> _run(BuildEnvironment environment, BuildOptions options,
+  Stream<BuildResult> _run(BuildEnvironment env, BuildOptions options,
```
Comment (natebosch) [inline build_runner/lib/src/generate/watch_impl.dart]: I don't think we should shorten this

File: build_runner/lib/src/generate/watch_impl.dart [hunk 3]
```diff
@@ -293,8 +319,11 @@ class WatchImpl implements BuildState {
           singleStepReader, _buildDefinition.assetGraph, optionalOutputTracker);
       _readerCompleter.complete(finalizedReader);
       _assetGraph = _buildDefinition.assetGraph;
-      build = await BuildImpl.create(_buildDefinition, options, buildPhases,
-          onDelete: _expectedDeletes.add);
+      build = await BuildImpl.create(
+        _buildDefinition,
+        options,
+        buildPhases,
```
Comment (natebosch) [inline build_runner/lib/src/generate/watch_impl.dart]: [nit] remove the trailing comma

File: build_runner/lib/src/generate/watch_impl.dart (hunk 2 above)
Comment (grouma) [reply build_runner/lib/src/generate/watch_impl.dart]: Didn't mean to. Was supposed to be temporary for static analysis errors. Fixed.

File: build_runner/lib/src/generate/build_definition.dart (hunk 1 above)
Comment (grouma) [reply build_runner/lib/src/generate/build_definition.dart]: Done.

File: build_runner/lib/src/generate/watch_impl.dart (hunk 3 above)
Comment (grouma) [reply build_runner/lib/src/generate/watch_impl.dart]: Done.

**Pull request merged**